File: bug71_incorrect_reverse_loop/bug71_incorrect_reverse_loop/main1.js
// Debugged JavaScript Code - Reverse loop fixed

function run() {
    let data = [];
    for (let i = 0; i < 10; i++)
        data.push(BigInt(0x100 * (i + 1)));

    let acc = 0n;
    let mask = 0xFFFFFFFFFFFFFFFFn;
    let CONST = 0x9E3779B97F4A7C15n;

    // FIX: include index 0
    for (let i = 9; i >= 0; i--) {
        acc ^= data[i] * BigInt(i + 1);
        acc = (acc * CONST) & mask;
        let r = BigInt(i % 7 + 1);
        acc = ((acc << r) | (acc >> (63n - BigInt(i % 7)))) & mask;
    }

    console.log("0x" + acc.toString(16));
}

run();

File: bug71_incorrect_reverse_loop/bug71_incorrect_reverse_loop/main1.java
public class main1 {
    public static void main(String[] args) {
        long[] data = new long[10];
        for (int i = 0; i < 10; i++)
            data[i] = 0x100L * (i + 1);

        long acc = 0;
        long mask = 0xFFFFFFFFFFFFFFFFL;

        // FIX: include index 0
        for (int i = 9; i >= 0; i--) {
            acc ^= data[i] * (i + 1);
            acc = (acc * 0x9E3779B97F4A7C15L) & mask;
            acc = ((acc << (i % 7 + 1)) | (acc >>> (63 - i % 7))) & mask;
        }

        System.out.println("0x" + Long.toHexString(acc));
    }
}